tools/lib/bpf/libbpf.c

```diff
@@ -7258,6 +7286,39 @@ static int bpf_object__resolve_externs(struct bpf_object *obj,
 	return 0;
 }
 
+static int __bpf_object__prepare(struct bpf_object *obj, int log_level,
+				 const char *target_btf_path)
+{
+	int err;
+
+	if (obj->prepared) {
+		pr_warn("object '%s': prepare can't be attempted twice\n", obj->name);
+		return libbpf_err(-EINVAL);
+	}
+
+	if (obj->gen_loader)
+		bpf_gen__init(obj->gen_loader, log_level);
```

Comment:
this got even better than before (in the old switch block to each relocation type). nice job.